src/main/kotlin/com/exactpro/th2/act/ActHandlerKt.kt

```diff
@@ -0,0 +1,389 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exactpro.th2.act
+
+import com.exactpro.th2.act.grpc.ActGrpc.ActImplBase
+import com.exactpro.th2.act.grpc.PlaceMessageRequest
+import com.exactpro.th2.act.grpc.PlaceMessageRequestOrBuilder
+import com.exactpro.th2.act.grpc.PlaceMessageResponse
+import com.exactpro.th2.act.impl.MessageResponseMonitor
+import com.exactpro.th2.act.receiver.AbstractMessageReceiver
+import com.exactpro.th2.act.receiver.MessageReceiver
+import com.exactpro.th2.act.rule.FieldCheckRuleKt
+import com.exactpro.th2.check1.grpc.Check1Service
+import com.exactpro.th2.check1.grpc.CheckpointRequest
+import com.exactpro.th2.common.event.Event
+import com.exactpro.th2.common.schema.message.MessageRouter
+import com.exactpro.th2.common.grpc.EventBatch
+import com.exactpro.th2.common.event.Event.Status.FAILED
+import com.exactpro.th2.common.event.Event.Status.PASSED
+import com.exactpro.th2.common.event.EventUtils.createMessageBean
+import com.exactpro.th2.common.grpc.Checkpoint
+import com.exactpro.th2.common.grpc.ConnectionID
+import com.exactpro.th2.common.grpc.Direction
+import com.exactpro.th2.common.grpc.EventID
+import com.exactpro.th2.common.grpc.Value.KindCase.SIMPLE_VALUE
+import com.exactpro.th2.common.grpc.Message
+import com.exactpro.th2.common.grpc.MessageBatch
+import com.exactpro.th2.common.grpc.MessageID
+import com.exactpro.th2.common.grpc.MessageMetadata
+import com.exactpro.th2.common.grpc.RequestStatus
+import com.exactpro.th2.common.grpc.Value
+import com.exactpro.th2.common.message.getField
+import com.exactpro.th2.common.message.messageType
+import com.exactpro.th2.common.message.toTreeTable
+import com.google.protobuf.TextFormat.shortDebugString
+import io.grpc.Context
+import io.grpc.Deadline
+import io.grpc.stub.StreamObserver
+import mu.KotlinLogging
+import org.apache.commons.lang3.BooleanUtils
+import org.apache.commons.lang3.exception.ExceptionUtils
+import java.time.Instant
+import java.util.concurrent.TimeUnit
+
+class ActHandlerKt(
+    private val messageRouter: MessageRouter<MessageBatch>,
+    private val subscriptionManager: SubscriptionManager,
+    private val eventRouter: MessageRouter<EventBatch>,
+    private val verifierConnector: Check1Service
+) : ActImplBase() {
+
+    override fun placeOrderFIX(
+        request: PlaceMessageRequest,
+        responseObserver: StreamObserver<PlaceMessageResponse>
+    ) {
+        runCatching {
+            LOGGER.debug { "placeOrderFIX request: $request}" }
```

Comment:
Please use TextFormat.shotDebug... or convert to JSON format to get a one-line value.